web_poet/rules.py

```diff
@@ -81,40 +81,33 @@ def __hash__(self):
         return hash((self.for_patterns, self.use, self.instead_of, self.to_return))
 
 
-class PageObjectRegistry(dict):
-    """This contains the :class:`~.ApplyRule` that associates the Page Objects
-    alongside its Items for a given URL matching rule.
-
-    PageObjectRegistry is a ``dict`` subclass with added functionalities on
```

Comment:
The idea here is to deprecate or remove `dict` API.

@BurnzZ - a couple of questions for you:

1.  would it cause issues for scrapy-poet?
2. would it be ok to remove it without deprecation?

Reply:
1. **scrapy-poet** wouldn't be affected at all as it's not relying on any `dict`-features of the registry. Moreover, its only interaction with the registry is via:

```python
SCRAPY_POET_OVERRIDES = default_registry.get_rules()
```

After retrieving the rules, it creates its own [OverridesRegistry](https://github.com/scrapinghub/scrapy-poet/blob/master/scrapy_poet/overrides.py#L28).

2. I think we should still retain the deprecation for backwards compatibility. I can imagine how users might be relying on the `dict`-interface for some of their needs.

If we remove it right away, there are some `dict` functionalities that would be left unattended:
- record deletion via key
- equality comparison (though `.get_rules()` could be used)
- length checks (though `.get_rules()` could be used)

Reply:
> I can imagine how users might be relying on the dict-interface for some of their needs.

I hope that the use of this feature is minimal now; I wouldn't expent many people to be using overrides registry at the moment, or customizing it.